starter-code/src/viking.js

```diff
@@ -1,11 +1,111 @@
 // Soldier
-function Soldier() {}
+function Soldier(health, strength) {
+    this.health=health;
+    this.strength=strength;
+}
+
+Soldier.prototype.attack=function(){
+    return this.strength;
+}
+Soldier.prototype.receiveDamage=function(damage){
+    this.health-=damage;
+}
 
 // Viking
-function Viking() {}
+function Viking(name,health,strength) {
+    Soldier.call(this,health,strength);
+    this.name=name;
+    
+}
+
+Viking.prototype = Object.create(Soldier.prototype);
+Viking.prototype.constructor = Viking;
+Viking.prototype.battleCry=function(){
+    return "Odin Owns You All!";
+}
 
+Viking.prototype.receiveDamage=function(damage){
+    this.health-=damage;
+    if(this.health-damage>0){
```

Comment:
This probably will not work because after you deduct the damage from the health you are checking if the new health - damage will be > 0 which means that you deducting double the damage.

Reply:
You're right! I did it right for saxon but not viking :)